app/src/main/java/com/github/se/eventradar/viewmodel/EventDetailsViewModel.kt

```diff
@@ -23,16 +28,33 @@ class EventDetailsViewModel
 @AssistedInject
 constructor(
     private val eventRepository: IEventRepository,
+    private val userRepository: IUserRepository,
     @Assisted val eventId: String,
 ) : ViewModel() {
 
   init {
     getEventData()
+    viewModelScope.launch {
+      when (val response = userRepository.getCurrentUserId()) {
+        is Resource.Success -> {
+          currentUserId = response.data
+        }
+        is Resource.Failure ->
+            Log.d(
+                "EventDetailsViewModel", "Could not get the user Id: ${response.throwable.message}")
+      }
+    }
   }
 
   private val _uiState = MutableStateFlow(EventUiState())
```

Comment:
I did this in another PR but these variable declarations need to be at the _top_ of the function. They cannot go after the `init` function.